rcl/test/rcl/test_wait.cpp

```diff
@@ -684,3 +685,97 @@ TEST_F(CLASSNAME(WaitSetTestFixture, RMW_IMPLEMENTATION), wait_set_get_allocator
   ret = rcl_wait_set_fini(&wait_set);
   EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
 }
+
+// Test wait set init failure cases using mocks
+TEST_F(CLASSNAME(WaitSetTestFixture, RMW_IMPLEMENTATION), wait_set_failed_init) {
+  rcl_wait_set_t wait_set = rcl_get_zero_initialized_wait_set();
+  // nullptr failures
+  rcl_ret_t ret =
+    rcl_wait_set_init(nullptr, 1, 1, 1, 1, 1, 0, context_ptr, rcl_get_default_allocator());
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret);
+  EXPECT_TRUE(rcl_error_is_set());
+  rcl_reset_error();
+
+  ret =
+    rcl_wait_set_init(&wait_set, 1, 1, 1, 1, 1, 0, nullptr, rcl_get_default_allocator());
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret);
+  EXPECT_TRUE(rcl_error_is_set());
+  rcl_reset_error();
+
+  {
+    // Mock rmw implementation to fail init
+    auto mock = mocking_utils::patch_and_return(
+      "lib:rcl", rmw_create_wait_set, nullptr);
+    ret =
+      rcl_wait_set_init(&wait_set, 1, 1, 1, 1, 1, 0, context_ptr, rcl_get_default_allocator());
+    EXPECT_EQ(RCL_RET_WAIT_SET_INVALID, ret);
+    EXPECT_TRUE(rcl_error_is_set());
+    rcl_reset_error();
+  }
+}
+
+// Test failure init when using a bad allocators
+TEST_F(CLASSNAME(WaitSetTestFixture, RMW_IMPLEMENTATION), wait_set_failed_init_bomb_alloc) {
+  rcl_wait_set_t wait_set = rcl_get_zero_initialized_wait_set();
+  rcl_ret_t ret = RCL_RET_OK;
+
+  // Pass bomb allocator to make resize function fail
+  rcl_allocator_t bomb_alloc = get_time_bombed_allocator();
+  for (int i = 0; i < 10; i++) {
+    set_time_bombed_allocator_count(bomb_alloc, i);
```

Comment:
@Blast545 consider using fault inject instead of the time bomb allocator. It has the same effect and it is more robust.

Reply:
Oh, there's `wait_set_test_maybe_init_fail`. I think we don't need this test.

Reply:
Agree, removed with c111d2a